### PR TITLE
feat: change secrets to parameters

### PR DESCRIPTION
**Pull request by christina-moore**
Have tested `fetch_secret` locally with hierarchical and non-hierarchical parameter names, but it feels a bit fragile so am open to suggestions on a better way forward!

### CODE REVIEW

**Comment by christina-moore** [comment]
Currently an ARN that doesn't match the regex will be logged as an `Unknown Error` - I can update this to be more verbose if it will help with clarity 🤔

**Comment by nick-solly** [comment]
How about we change the terraform input parameter to be the Parameter Name rather than the ARN?  Then it can be passed straight through to the get_parameter function without needing to regex or worry about hierarchical or non-hierarchical params?

**Comment by christina-moore** [comment]
> How about we change the terraform input parameter to be the Parameter Name rather than the ARN? Then it can be passed straight through to the get_parameter function without needing to regex or worry about hierarchical or non-hierarchical params?

Hm we need the ARN for the IAM policy to access the parameter though: https://github.com/nick-solly/knimer/pull/1/files#diff-27ba68ec068c9a6ec6507d449953b624601d4fbadf7cc1f7972d81cd79f48205R41

**Comment by christina-moore** [comment]
> > How about we change the terraform input parameter to be the Parameter Name rather than the ARN? Then it can be passed straight through to the get_parameter function without needing to regex or worry about hierarchical or non-hierarchical params?
> 
> Hm we need the ARN for the IAM policy to access the parameter though: https://github.com/nick-solly/knimer/pull/1/files#diff-27ba68ec068c9a6ec6507d449953b624601d4fbadf7cc1f7972d81cd79f48205R41

I've added a parameter name so users will need to pass both name and ARN

File: terraform/slack-trigger/variables.tf [hunk 1]
```diff
@@ -3,8 +3,13 @@ variable "name_prefix" {
   type        = string
 }
 
+variable "slack_signing_secret_name" {
+  description = "Name of the parameter containing the slack signing secret"
+  type        = string
+}
+
 variable "slack_signing_secret_arn" {
```
Comment (nick-solly) [inline terraform/slack-trigger/variables.tf:11]: is the `slack_signing_secret_arn` needed for the slack-trigger or can this block be removed?

File: terraform/knimer/variables.tf [hunk 2]
```diff
@@ -42,7 +42,7 @@ variable "workflow_secrets" {
 }
```
Comment (nick-solly) [inline terraform/knimer/variables.tf:43]: does the `slack_webhook_url_secret_name` need adding to this file and then passing in `main.tf` to the `slack` module?

File: terraform/knimer/slack/variables.tf [hunk 3]
```diff
@@ -3,8 +3,13 @@ variable "name_prefix" {
   type        = string
 }
 
+variable "slack_webhook_url_secret_name" {
+  description = "Name of the parameter containing the slack webhook url"
+  type        = string
+}
+
 variable "slack_webhook_url_secret_arn" {
```
Comment (nick-solly) [inline terraform/knimer/slack/variables.tf:11]: does the `lambda` module need `slack_webhook_url_secret_arn`?

**Review by nick-solly — CHANGES_REQUESTED** [review]

File: terraform/slack-trigger/variables.tf (hunk 1 above)
Comment (christina-moore) [reply terraform/slack-trigger/variables.tf:11]: ah yes it's needed to build the IAM role to access it https://github.com/nick-solly/knimer/pull/1/files/1eea4013bc9b0f7e44c82f664b74632dde3a9686#diff-27ba68ec068c9a6ec6507d449953b624601d4fbadf7cc1f7972d81cd79f48205R41

File: terraform/knimer/variables.tf (hunk 2 above)
Comment (christina-moore) [reply terraform/knimer/variables.tf:43]: good spot! https://github.com/nick-solly/knimer/pull/1/commits/34254670b336631246f4604fd6c4a7ca4da33087

File: terraform/knimer/slack/variables.tf (hunk 3 above)
Comment (christina-moore) [reply terraform/knimer/slack/variables.tf:11]: it's still needed for the `lambda-secret-access-policy`

**Review by nick-solly — DISMISSED** [review]

**Review by nick-solly — APPROVED** [review]

**Pull request merged**